app.py

```diff
@@ -36,8 +43,90 @@ def datasets(index: str):
 @app.route('/datasets/refresh')
 def refresh_datasets():
     global masader
+    global embeddings
+    global cluster
 
     print('Refreshing the dataset...')
-    masader = load_masader_dataset_as_dict()
+    data = list(load_masader_dataset_as_dict())
+    masader = data
+    embeddings = get_embeddings_data(data)
+    cluster = get_cluster_data()
 
     return jsonify(f'The datasets updated successfully! The current number of available datasets is {len(masader)}.')
+
+
+# -----------Clusters-------------
+
+@app.route('/cluster')
+def cluster_req():
+    return jsonify(cluster)
+
+
+def get_cluster_data():
+    global tsne_data
+
+    clustering = KMeans(n_clusters=15).fit(tsne_data)
+    return pd.DataFrame(clustering.labels_.reshape(len(tsne_data), 1)).to_json()
+
+
+# -----------Embeddings-------------
+
+@app.route('/embeddings')
+def embeddings_req():
+    return jsonify(embeddings)
+
+
+def get_embeddings_data(dataset):
+    desc = [desc if desc != 'nan' else "" for desc in dataset['Description']]
+    tasks = [tasks if tasks != 'nan' else "" for tasks in dataset['Tasks']]
+    abstracts = [abs if abs != 'nan' else "" for abs in dataset['Abstract']]
+
+    full = [" ".join(tpl) for tpl in list(zip(*[desc, tasks, abstracts]))]
+
+    return embeddings_data(sentence_transformer_request(full))
+
+
+def sentence_transformer_request(d_abstracts):
+    requested_data = None
+    first = 0
+    last = 0
+
+    print(len(d_abstracts))
+
+    while last < len(d_abstracts):
+
+        if last + 200 < len(d_abstracts):
+            last += 200
+        else:
+            last = len(d_abstracts)
+
+        response = requests.post(API_URL + "sentence-transformers/all-MiniLM-L6-v2", headers=headers,
```

Comment:
I'm using black to format the code with maximum line length of 120 and single quotes instead of double, could you please use the same settings for have similar styles?